src/test/java/org/galatea/starter/entrypoint/HalRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+package org.galatea.starter.entrypoint;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import feign.Feign;
+import feign.Param;
+import feign.RequestLine;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@RequiredArgsConstructor
+@Slf4j
+@ToString
+@EqualsAndHashCode
+@Category(org.galatea.starter.IntegrationTestCategory.class)
+public class HalRestControllerIntegrationTest {
+
+  @Test
+  public void testCoinFlip() {
+    String fuseHostName = System.getProperty("fuse.sandbox.url");
+    if (fuseHostName == null || fuseHostName.isEmpty()) {
+      // TODO: the base URL should probably be moved to a src/test/resources properties file
```

Comment:
Sonar will complain about the new TODO here. TODOs are flagged as code debt and we have rules against adding those.

Reply:
- [ ] Remove todos
- [ ] Move base URL to a resource file